go/vt/dbconfigs/dbconfigs.go

```diff
@@ -113,64 +115,47 @@ type UserConfig struct {
 	UseTCP   bool   `json:"useTcp,omitempty"`
 }
 
-// RegisterFlags registers the flags for the given DBConfigFlag.
-// For instance, vttablet will register client, dba and repl.
-// Returns all registered flags.
+// RegisterFlags registers the base DBFlags, credentials flags, and the user
+// specific ones for the specified system users for the requesting command.
+// For instance, the vttablet command will register flags for all users
+// as defined in the dbconfigs.All variable.
 func RegisterFlags(userKeys ...string) {
-	registerBaseFlags()
-	for _, userKey := range userKeys {
-		uc, cp := GlobalDBConfigs.getParams(userKey, &GlobalDBConfigs)
-		registerPerUserFlags(userKey, uc, cp)
-	}
-}
-
-func registerBaseFlags() {
-	flag.StringVar(&GlobalDBConfigs.Socket, "db_socket", "", "The unix socket to connect on. If this is specified, host and port will not be used.")
-	flag.StringVar(&GlobalDBConfigs.Host, "db_host", "", "The host name for the tcp connection.")
-	flag.IntVar(&GlobalDBConfigs.Port, "db_port", 0, "tcp port")
-	flag.StringVar(&GlobalDBConfigs.Charset, "db_charset", "utf8mb4", "Character set used for this tablet.")
-	flag.Uint64Var(&GlobalDBConfigs.Flags, "db_flags", 0, "Flag values as defined by MySQL.")
-	flag.StringVar(&GlobalDBConfigs.Flavor, "db_flavor", "", "Flavor overrid. Valid value is FilePos.")
-	flag.Var(&GlobalDBConfigs.SslMode, "db_ssl_mode", "SSL mode to connect with. One of disabled, preferred, required, verify_ca & verify_identity.")
-	flag.StringVar(&GlobalDBConfigs.SslCa, "db_ssl_ca", "", "connection ssl ca")
-	flag.StringVar(&GlobalDBConfigs.SslCaPath, "db_ssl_ca_path", "", "connection ssl ca path")
-	flag.StringVar(&GlobalDBConfigs.SslCert, "db_ssl_cert", "", "connection ssl certificate")
-	flag.StringVar(&GlobalDBConfigs.SslKey, "db_ssl_key", "", "connection ssl key")
-	flag.StringVar(&GlobalDBConfigs.TLSMinVersion, "db_tls_min_version", "", "Configures the minimal TLS version negotiated when SSL is enabled. Defaults to TLSv1.2. Options: TLSv1.0, TLSv1.1, TLSv1.2, TLSv1.3.")
-	flag.StringVar(&GlobalDBConfigs.ServerName, "db_server_name", "", "server name of the DB we are connecting to.")
-	flag.IntVar(&GlobalDBConfigs.ConnectTimeoutMilliseconds, "db_connect_timeout_ms", 0, "connection timeout to mysqld in milliseconds (0 for no timeout)")
-	flag.BoolVar(&GlobalDBConfigs.EnableQueryInfo, "db_conn_query_info", false, "enable parsing and processing of QUERY_OK info fields")
+	servenv.OnParse(func(fs *pflag.FlagSet) {
+		registerBaseFlags(fs)
+		for _, userKey := range userKeys {
+			uc, cp := GlobalDBConfigs.getParams(userKey, &GlobalDBConfigs)
```

Comment:
i don't have a particular answer in mind, but the question occurred to me: for both the base flags (on `GlobalDBConfigs`) and the per-user flags, does it make sense to use the variable as the default value?

by which i mean, instead of `fs.StringVar(&GlobalDBConfigs.Socket, "db_socket", "", "the unix socket ...")` we do `fs.StringVar(&GlobalDBConfigs.Socket, "db_socket", GlobalDBConfigs.Socket, "the unix socket ...")` and so on?

Reply:
Good catch. That's what I've been doing, but overlooked it here. Will update it.